hera_cal/apply_cal.py

```diff
@@ -308,7 +307,9 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
                         if np.all(data_flags[bl]):
                             redundant_weights[bl][:] = 0.
                 # redundantly average
-                utils.red_average(data=data, flags=data_flags, nsamples=data_nsamples, reds=all_red_antpairs, wgts=redundant_weights, inplace=True)
+                utils.red_average(data=data, flags=data_flags, nsamples=data_nsamples,
+                                  reds=all_red_antpairs, wgts=redundant_weights, inplace=True,
+                                  propagate_flags=True)
```

Comment:
@nkern, @jsdillon  and I decided that putting the redundant averaging inside of `apply_cal` was the best option to deal with partial i/o. We need to be able to redundantly average partially loaded chunks of baselines that have just had their calibration solution applied before partially writing them and loading another chunk. I don't see how we can do this in a completely separate RTP step without saving the calibrated chunks into a bunch of small files which I think would be much messier.

Reply:
yeah it seemed like an RTP consideration. I think the best thing to do is, instead of adding a feature to apply_cal that should be in red_average, is to add it to red_average, and then just call red_average from within the apply_cal drive script. Does that make sense?

Reply:
@nkern, it does seem that we would need a wrapper for `red_average` to do all the i/o.

Reply:
Partial `i/o` is a generally useful feature for `red_average` to have so I'm open to this. Thoughts @jsdillon ?

Reply:
On the other hand, moving the `i/o` into `red_average` (or some wrapper) is a lot of work for a use case that hasn't yet materialized. We have something that works for our use case. It might not be the most optimal way to organize the code but in the interest of getting other stuff working, I personally would like to move on. I'm happy to write an issue for this.

Reply:
Yeah I'm fine with that for now.

Reply:
Cool!